package/MDAnalysis/analysis/base.py

```diff
@@ -447,8 +494,177 @@ def run(self, start=None, stop=None, step=None, frames=None,
+            number of workers (local or remote processes).
+        """
+        if scheduler == 'localdask':
+            n_workers = 1
+        else:
+            from dask.distributed import Client
+            if isinstance(scheduler, Client):
+                # TODO: add assertions that make sure that 
+                # you don't set 
+                # both `scheduler` and `n_workers/threads_per_worker`
+                n_workers = len(scheduler.ncores())
+            else:
+                kwargs = {'n_workers':n_workers}
+            self._scheduler_kwargs = kwargs
+        self._n_bslices = self._n_workers = n_workers
+
+    def run(self, start=None, stop=None, step=None, frames=None,
+            verbose=None, n_workers=1, 
+            *, scheduler=None,
+            progressbar_kwargs={},
+            ):
+        """Perform the calculation
+
+        Parameters
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+            .. versionadded:: 2.2.0
+
+        verbose : bool, optional
+            Turn on verbosity
+        
+        scheduler : str, optional
+            Enables running with different schedulers.
+        
+        progressbar_kwargs : dict, optional
+            ProgressBar keywords with custom parameters regarding progress bar position, etc; 
+            see :class:`MDAnalysis.lib.log.ProgressBar` for full list.
+
+
+        .. versionchanged:: 2.2.0
+            Added ability to analyze arbitrary frames by passing a list of
+            frame indices in the `frames` keyword argument.
+
+        .. versionchanged:: 2.5.0
+            Add `progressbar_kwargs` parameter, 
+            allowing to modify description, position etc of tqdm progressbars
+        """
+        if scheduler is None: # fallback to the local scheduler
+            self._compute(start=start, stop=stop, step=step, frames=frames, 
+            verbose=verbose, progressbar_kwargs=progressbar_kwargs)
+        else:
+            self._setup_scheduler(scheduler=scheduler, n_workers=n_workers)
+            if scheduler == 'localdask': # imitation of dask mainly for testing purposes
+                delayed = localdelayed
+            else: # elif scheduler is a type of dask scheduler
+                try:
+                    from dask.delayed import delayed
+                except ImportError:
+                    def delayed(*args, **kwargs): # need implementation to avoid syntax warnings below
```

Comment:
Where would syntax warnings come from? From Python??

Remove — it's noisy code. If you need to teach your IDE not to warn you then you have to do it in a way that doesn't introduce junk code.